### PR TITLE
Scraper integration

### PR DESCRIPTION
**Pull request by juliusBaranauskas**
Final (almost) webscraper integration
- Now scraper gathers universities (Lithuania's only to not destroy the first launch time for everyone)
- There shouldn't be no changes in database structure so no need to delete or alter databases, should work as it is without overwriting your current database entries.

If you want to update the universities to the ones gathered from the web, follows these steps:

1. Delete current database in SSMS
2. Launch the project (it should throw some exception about missing object "dbo...")
3. Stop the project.
4. In SSMS refresh the database and if new tables for user logins appeared, delete them EXCEPT __MigrationHistory, it should be the only one left in database.
5. In VS Tools/NuGet PackageManager/Console type in `Update-Database`
6. If it doesn't throw any errors, then launch the project and it should launch eventually with new database entries

### CODE REVIEW

File: Webserver/Data/FillDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using Models;
+using System.Linq;
+using WebScraper;
+using System;
+
+namespace Webserver.Data
+{
+    public static class FillDatabase
```
Comment (Doombringer-MarAlith) [inline Webserver/Data/FillDatabase.cs]: Manau reiktų parefactorint čia:
1. Šitą klasę pervadinam į DatabaseFiller.cs
2. Pakeičiam i non-static, per DI pasiimam `Scraper` fieldą
2. Perkeliam klasę į naują folderį `Data/Services`
3. Pridedam šitos klasės fieldą į `DatabaseFactory.cs` per DI
4. `DatabaseFactory::Initialize()` metode paleidžiam (dar ten pat per `ApplicationDbContext` reikia patikrint, ar yra bent vienas uni/fac/prog duombazėj, jei taip, tai nepaleidžiam

File: Webserver/Data/FillDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                    if (!dbContext.Faculties.Any())
+                    {
+                        foreach (var currentCount in facultyCount)
+                        {
+                            for (int i = 0; i < currentCount; i++)
+                            {
+                                faculties[currentIndex].UniversityId = id;
+                                dbContext.Faculties.Add(faculties[currentIndex]);
+                                currentIndex++;
+                            }
```
Comment (Doombringer-MarAlith) [inline Webserver/Data/FillDatabase.cs]: Newline

File: Webserver/Data/FillDatabase.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    if (!dbContext.Programmes.Any())
+                    {
+                        foreach (var currentCount in programmeCount)
+                        {
+                            for (int i = 0; i < currentCount; i++)
+                            {
+                                programmes[currentIndex].FacultyId = id;
+                                dbContext.Programmes.Add(programmes[currentIndex]);
+                                currentIndex++;
+                            }
```
Comment (Doombringer-MarAlith) [inline Webserver/Data/FillDatabase.cs]: Newline

File: Webserver/Data/FillDatabase.cs (hunk 1 above)
Comment (juliusBaranauskas) [reply Webserver/Data/FillDatabase.cs]: Man DatabaseFactory per Initialize parametrus pridėt tą savo DatabaseFiller?

Comment (Doombringer-MarAlith) [reply Webserver/Data/FillDatabase.cs]: Ne, susikurk konstruktorių ir per DI kaip klasės fieldą

File: Webserver/Data/ApplicationDbContext.cs [hunk 4]
```diff
@@ -4,6 +4,8 @@
 using Webserver.Data.Configuration;
 using Webserver.Data.Models;
 using Webserver.Models;
+using WebScraper;
+
```
Comment (Doombringer-MarAlith) [inline Webserver/Data/ApplicationDbContext.cs]: Remove line

File: Webserver/Data/Infrastructure/DatabaseFactory.cs [hunk 5]
```diff
@@ -1,8 +1,26 @@
...
 {
     public class DatabaseFactory : IDatabaseFactory
     {
         private ApplicationDbContext _dbContext;
+        private DatabaseFiller _dbFiller;
+        private IGatherDatabase _scraper;
+
+        public DatabaseFactory(DatabaseFiller dbFiller, IGatherDatabase scraper)
+        {
+            if (_dbFiller == null)
```
Comment (Doombringer-MarAlith) [inline Webserver/Data/Infrastructure/DatabaseFactory.cs]: Šitų nereikia check'ų, visada bus null

**Pull request merged**